### PR TITLE
Suspend user

### PR DESCRIPTION
**Pull request by alinhle**
## What does this PR do?
- Implements the new Suspend User Story [#144459753](https://www.pivotaltracker.com/story/show/144459753)
- Implements a [notification service](https://github.com/coralproject/talk/pull/600/files#diff-539b42b1f378342ce606737100fab9e0)

## How do I test this PR?

- Suspend any user in the mod queue
- Check if email has been received
- Login as the suspended user, see suspension note, and all mutations should fail.
- Wait until suspension time is over, and see if suspension has been lifted :-D

### CODE REVIEW

File: models/user.js [hunk 1]
```diff
@@ -212,11 +221,12 @@ UserSchema.method('can', function(...actions) {
     throw new Error(`invalid actions: ${actions}`);
   }
 
-  if (this.status === 'BANNED') {
+  if (this.status === 'BANNED' || this.suspension.until && this.suspension.until > new Date()) {
```
Comment (wyattjoh) [inline models/user.js]: You should wrap this conditional up:

```diff
- if (this.status === 'BANNED' || this.suspension.until && this.suspension.until > new Date()) {
+ if (this.status === 'BANNED' || (this.suspension.until && this.suspension.until.getTime() > Date.now())) {
```

File: services/users.js [hunk 2]
```diff
@@ -450,41 +450,77 @@ module.exports = class UsersService {
...
+          },
+        }
+      })
+      .then((user) => {
+        if (message) {
+          let localProfile = user.profiles.find((profile) => profile.provider === 'local');
+
+          if (localProfile) {
+            const options =
+              {
```
Comment (wyattjoh) [inline services/users.js]: Move this up a line as well

File: services/users.js [hunk 3]
```diff
@@ -450,41 +450,77 @@ module.exports = class UsersService {
...
-              to: localProfile.id  // This only works if the user has registered via e-mail.
-                                   // We may want a standard way to access a user's e-mail address in the future
-            };
-
-          return MailerService.sendSimple(options);
-        } else {
-          return Promise.reject(errors.ErrMissingEmail);
+  static rejectUsername(id, message) {
+    return UserModel.findOneAndUpdate(
+      {id}, {
```
Comment (wyattjoh) [inline services/users.js]: Same as comment above

File: services/users.js [hunk 4]
```diff
@@ -450,41 +450,77 @@ module.exports = class UsersService {
...
         }
-      }
-    });
+      }).then((user) => {
+        if (message) {
+          let localProfile = user.profiles.find((profile) => profile.provider === 'local');
+
+          if (localProfile) {
+            const options =
+              {
```
Comment (wyattjoh) [inline services/users.js]: Same as comment above

**Review by riley — APPROVED** [review]
LGTM! We're going to have to coordinate on merging either this one or the Multiple Roles PR first, because it's going to cause conflicts.

**Pull request merged**